Assets/__Scripts/MapEditor/Grid/Collections/BeatmapObjectContainerCollection.cs

```diff
@@ -237,7 +237,9 @@ public void RemoveConflictingObjects(IEnumerable<BeatmapObject> newObjects, out
         foreach (BeatmapObject newObject in newObjects)
         {
             Debug.Log($"Performing conflicting check at {newObject._time}");
-            BeatmapObject conflict = UnsortedObjects.Find(x => x.IsConflictingWith(newObject));
+
+            var localWindow = GetBetween(newObject._time - 0.1f, newObject._time + 0.1f);
+            BeatmapObject conflict = localWindow.FirstOrDefault(x => x.IsConflictingWith(newObject));
```

Comment:
last i've checked, a straight `FirstOrDefault(predicate)` check is slower than a `.Where(predicate).FirstOrDefault()`. Does that apply here? I'm pretty sure the former does a bit extra processing to ensure that the match it found really is the first in the collection.

Reply:
localWindow should be small enough for this to not really matter but I've changed it anyway